### PR TITLE
Clean cross entropy

### PR DESCRIPTION
**Pull request by reyoung**
Unify cross entropy CPU and GPU kernel by `platform::ForRange` functor. It reduce the lines of code of `cross_entropy_op`

### CODE REVIEW

File: paddle/fluid/operators/cross_entropy_op.h [hunk 1]
```diff
@@ -17,69 +17,106 @@ limitations under the License. */
...
+ public:
+  XeGradFunctor(T* dx,
+                const T* dy,           // NOLINT
+                const T* x,            // NOLINT
+                const int64_t* label,  // NOLINT
+                size_t num_classes)
+      : dx_(dx), dy_(dy), x_(x), label_(label), num_classes_(num_classes) {}
+
+  HOSTDEVICE void operator()(size_t label_id) {
+    auto x_is_true_offset = label_id * num_classes_ + label_[label_id];
```
Comment (chengduoZH) [inline paddle/fluid/operators/cross_entropy_op.h]: Maybe `label_id` is not exact, `sample_id` may be better.

File: paddle/fluid/operators/cross_entropy_op.h [hunk 2]
```diff
@@ -17,69 +17,106 @@ limitations under the License. */
...
+    for (size_t x_offset = label_id * num_classes_;
+         x_offset < (label_id + 1) * num_classes_; ++x_offset) {
+      dx_[x_offset] = x_offset != x_is_true_offset
+                          ? static_cast<T>(0)
+                          : -dy_[label_id] / x_[x_offset];
+    }
```
Comment (chengduoZH) [inline paddle/fluid/operators/cross_entropy_op.h]: I prefer to write like this:
```
auto x_is_true_offset = label_id * num_classes_ + label_[label_id];
dx_[x_is_true_offset] = -dy_[label_id] / x_[x_is_true_offset]
```
and put setting `dx_` to zero in [here](https://github.com/PaddlePaddle/Paddle/pull/10280/files#diff-a62bef624c671424afbf8ae28466d07eR116).

Comment (reyoung) [reply paddle/fluid/operators/cross_entropy_op.h]: It is slower to use two kernels than one kernel.

File: paddle/fluid/operators/cross_entropy_op.h (hunk 1 above)
Comment (reyoung) [reply paddle/fluid/operators/cross_entropy_op.h]: Done.

**Pull request merged**